### ISSUE TITLE
MADT PCAT_COMPAT flag values reversed?

### ISSUE THREAD

**Issue by Jnosh** [acpica/acpica#224]
In `source/include/actbl1.h`, `ACPI_TABLE_MADT` includes a flag field.

The preprocessor define `ACPI_MADT_PCAT_COMPAT` specifies a mask for the PCAT_COMPAT flag while `ACPI_MADT_DUAL_PIC` and `ACPI_MADT_MULTIPLE_APIC` specify the values of said flag.

My interpretation of said defines is that `ACPI_MADT_DUAL_PIC` indicates the presence of dual 8259 PICs, whereas `ACPI_MADT_MULTIPLE_APIC` indicates their absence.

i.e. `hasDualPics = (madt->Flags & ACPI_MADT_PCAT_COMPAT) == ACPI_MADT_DUAL_PIC`

However, `ACPI_MADT_DUAL_PIC` is defined as `0` while the ACPI specification says regarding the PCAT_COMPAT flag: 
> A one indicates that the system also has a PC-AT-compatible dual-8259 setup.
> (ACPI 6.1, Table 5-45 Multiple APIC Flags, p. 144)

I am probably misunderstanding or misreading something but I would appreciate any insight.

**Comment by acpibob** [comment]
Looks like you might be right

**Comment by Jnosh** [comment]
Thanks. I am currently running on Qemu, Bochs and Virtualbox, all of which emulate the PICs and they report `1` as the value of that flag in their MADTs. I was worried I might be misunderstanding how to use the defined constants.

I added a pull request to fix this, if that is OK: https://github.com/acpica/acpica/pull/225

I hope this doesn't cause any compatibility issues but since no-one noticed that before, those definitions probably aren't in broad use?
I did some quick googling and only found the definitions in the ACPICA header in various OSS projects but no actual uses.
